### PR TITLE
block and unblock user

### PR DESCRIPTION
**Pull request by Valdenirmezadri**

### CODE REVIEW

File: user.go [hunk 1]
```diff
@@ -519,3 +519,46 @@ func (cli *Client) usync(ctx context.Context, jids []types.JID, mode, context st
...
 	}
 }
+
+func (cli *Client) UnblockUser(target types.JID) (Blockedlist []types.JID, err error) {
+	return cli.updateBlock(target, false)
+}
+
+func (cli *Client) BlockUser(target types.JID) (Blockedlist []types.JID, err error) {
+	return cli.updateBlock(target, true)
+}
```
Comment (tulir) [inline user.go:529]:
```suggestion
func (cli *Client) UnblockUser(target types.JID) ([]types.JID, error) {
	return cli.updateBlock(target, false)
}

func (cli *Client) BlockUser(target types.JID) ([]types.JID, error) {
	return cli.updateBlock(target, true)
}
```

That return variable name is wrong (shouldn't start with uppercase) and it's unnecessary anyway

File: user.go [hunk 2]
```diff
@@ -519,3 +519,46 @@ func (cli *Client) usync(ctx context.Context, jids []types.JID, mode, context st
...
+		Attrs: waBinary.Attrs{"action": action, "jid": target},
+	}}
+	resp, err := cli.sendIQ(infoQuery{
+		Namespace: "blocklist",
+		Type:      iqSet,
+		To:        types.ServerJID,
+		Content:   content,
+	})
+	if err != nil {
+		return []types.JID{}, err
```
Comment (tulir) [inline user.go:549]:
```suggestion
		return nil, err
```

**Review by tulir — COMMENTED** [review]
Should there also be a method to get the current blocklist? 🤔

**Comment by Valdenirmezadri** [comment]
> Should there also be a method to get the current blocklist? thinking

should, but unfortunately I haven't found a way to request the list

**Comment by Stein-Hakase** [comment]
> > Should there also be a method to get the current blocklist? thinking
> 
> should, but unfortunately I haven't found a way to request the list

Maybe you can just fake an unblock request ( by giving any phone number or try it empty ), if success the response should be always the current blocked list!

**Pull request closed without merging**